lib/apiservers/engine/backends/container.go

```diff
@@ -303,7 +303,12 @@ func (c *Container) ContainerCreate(config types.ContainerCreateConfig) (types.C
 
 		res, err := client.Storage.VolumeJoin(joinParams)
 		if err != nil {
-			return types.ContainerCreateResponse{}, derr.NewErrorWithStatusCode(fmt.Errorf("Server error from Portlayer: %s", err), http.StatusInternalServerError)
+			switch err := err.(type) {
+			case *storage.VolumeJoinInternalServerError:
+				return nil, derr.NewErrorWithStatusCode(err.Payload.Message, http.StatusInternalServerError)
+			case *storage.VolumeJoinDefault:
+				return nil, derr.NewErrorWithStatusCode(err.Payload.Message, http.StatusInternalServerError)
+			}
```

Comment:
ah there are going to be a lot of these.... I will look through them and fix it.